crates/polars-plan/src/dsl/expr.rs

```diff
@@ -154,7 +170,7 @@ pub enum Expr {
         output_type: GetOutput,
         options: FunctionOptions,
     },
-    SubPlan(Box<LogicalPlan>, Vec<String>),
+    SubPlan(Box<SubLogicalPlan>, Vec<String>),
```

Comment:
Can you use `SpecialEq` here. That already defaults to `false`.

Reply:
How's this? Or should `LogicalPlan` still be wrapped by a `SubLogicalPlan`